lib/elixir/lib/string_io.ex

```diff
@@ -263,37 +257,21 @@ defmodule StringIO do
 
   ## get_line
 
-  defp get_line(encoding, prompt,
-                %{input: input, output: output, capture_prompt: capture_prompt} = s) do
-    case :unicode.characters_to_list(input, encoding) do
-      {:error, _, _} ->
-        {{:error, :collect_line}, s}
-      {:incomplete, _, _} ->
-        {{:error, :collect_line}, s}
-      chars ->
-        {result, input} = do_get_line(chars, encoding)
+  defp get_line(encoding, prompt, %{input: input} = s) do
+    with {result, input} <- binary_collect(input, encoding, "") do
+      result = case result do
+        "" -> :eof
+        r -> r
+      end
 
-        s =
-          if capture_prompt do
-            %{s | output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
-          else
-            s
-          end
+      s = capture_prompt(s, prompt)
 
-        {result, %{s | input: input}}
+      {result, %{s | input: input}}
```

Comment:
As you're already refactoring it for performance purposes, why not sending `input` as a parameter for capture prompt and doing this mapset together with the `output` attribute in the [`capture_prompt\2`](https://github.com/elixir-lang/elixir/pull/6056/files#diff-b27997fe2e28aaac3723720c70523ea1R342) function?

This is also duplicated code with [`get_chars\4`](https://github.com/elixir-lang/elixir/pull/6056/files#diff-b27997fe2e28aaac3723720c70523ea1R226)